Sources/Datadog/FeaturesIntegration/WebView/WebRUMEventConsumer.swift

```diff
@@ -6,8 +6,92 @@
 
 import Foundation
 
+// TODO: RUMM-1786 implement mappers
+internal class WebRUMEventMapper { }
+
 internal class WebRUMEventConsumer: WebEventConsumer {
-    func consume(event: [String: Any], eventType: String) {
-        // TODO: RUMM-1791 implement event consumers
+    private let dataWriter: Writer?
+    private let dateCorrector: DateCorrectorType?
+    private let webRUMEventMapper: WebRUMEventMapper?
+    private let contextProvider: RUMContextProvider?
+
+    init(
+        dataWriter: Writer?,
+        dateCorrector: DateCorrectorType?,
+        webRUMEventMapper: WebRUMEventMapper?,
+        contextProvider: RUMContextProvider?
```

Comment:
Because of these optionals, we must struggle with optionality in this class implementation. I think it doesn't make sense to instantiate `WebRUMEventConsumer` if we don't have any of these values. Also, each optional adds to the number of possible code execution branches, making it more error prone.